js/navbar/tabColor.js

```diff
@@ -1,149 +1,123 @@
-      }
-      if (pixels[k] > ct) {
-        largestPixelSet = k
-        ct = pixels[k]
-      }
+    if (sum < 50) {
+      add = 0.01
     }
 
-    var res = largestPixelSet.split(',')
-
-    for (var i = 0; i < res.length; i++) {
-      res[i] = parseInt(res[i])
+    if (data[i] > 210 || data[i + 1] > 210 || data[i + 2] > 210) {
+      add = 0.5 - (0.0001 * sum)
     }
 
-    callback(res)
+    if (pixels[d]) {
+      pixels[d] = pixels[d] + add
+    } else {
+      pixels[d] = add
+    }
   }
 
-  colorExtractorImage.src = url
-}
-
-var colorExtractorImage = document.createElement('img')
+  // find the largest pixel set
+  var largestPixelSet = null
+  var ct = 0
 
-const defaultColors = {
-  private: ['rgb(58, 44, 99)', 'white'],
-  lightMode: ['rgb(255, 255, 255)', 'black'],
-  darkMode: ['rgb(40, 44, 52)', 'white']
-}
+  for (var k in pixels) {
+    if (k === '255,255,255' || k === '0,0,0') {
+      pixels[k] *= 0.05
+    }
+    if (pixels[k] > ct) {
+      largestPixelSet = k
+      ct = pixels[k]
+    }
+  }
 
-var hours = new Date().getHours() + (new Date().getMinutes() / 60)
+  var res = largestPixelSet.split(',')
 
-// we cache the hours so we don't have to query every time we change the color
+  for (var i = 0; i < res.length; i++) {
+    res[i] = parseInt(res[i])
+  }
 
-setInterval(function () {
-  var d = new Date()
-  hours = d.getHours() + (d.getMinutes() / 60)
-}, 4 * 60 * 1000)
-
-function updateTabColor (favicons, tabId) {
-  // private tabs always use a special color, we don't need to get the icon
-  if (tabs.get(tabId).private === true) {
-    return
+  // dim the colors late at night or early in the morning, or when dark mode is enabled
+  var colorChange = 1
+  if (hours > 20) {
+    colorChange -= 0.015 * Math.pow(2.75, hours - 20)
+  } else if (hours < 6.5) {
+    colorChange -= -0.15 * Math.pow(1.36, hours) + 1.15
   }
-  requestIdleCallback(function () {
-    getColor(favicons[0], function (c) {
-      // dim the colors late at night or early in the morning, or when dark mode is enabled
-      var colorChange = 1
-      if (hours > 20) {
-        colorChange -= 0.015 * Math.pow(2.75, hours - 20)
-      } else if (hours < 6.5) {
-        colorChange -= -0.15 * Math.pow(1.36, hours) + 1.15
-      }
 
-      if (window.isDarkMode) {
-        colorChange = Math.min(colorChange, 0.58)
-      }
+  if (window.isDarkMode) {
+    colorChange = Math.min(colorChange, 0.58)
+  }
 
-      c[0] = Math.round(c[0] * colorChange)
-      c[1] = Math.round(c[1] * colorChange)
-      c[2] = Math.round(c[2] * colorChange)
+  res[0] = Math.round(res[0] * colorChange)
+  res[1] = Math.round(res[1] * colorChange)
+  res[2] = Math.round(res[2] * colorChange)
 
-      var cr = 'rgb(' + c[0] + ',' + c[1] + ',' + c[2] + ')'
+  return res
+}
 
-      var obj = {
-        r: c[0] / 255,
-        g: c[1] / 255,
-        b: c[2] / 255
-      }
+function getHours () {
+  return new Date().getHours() + (new Date().getMinutes() / 60)
```

Comment:
If you're making a function out of it anyway, you can accept an optional `date` parameter to avoid instantiation `new Date()` twice:

    function getHours (date = new Date()) {
      return date.getHours() + date.getMinutes() / 60
    }

Reply:
Or even:

    const getHours = (date = new Date()) => date.getHours() + date.getMinutes() / 60;